### PR TITLE
Aztec v1.0-beta.4

### LINKED ISSUE #6322: Enable Aztec for beta and release users

### PR DESCRIPTION
**Pull request by 0nko**
Resolves #6322 and includes the latest Aztec release.

### CODE REVIEW

File: WordPress/build.gradle [hunk 1]
```diff
@@ -51,7 +50,6 @@ android {
 
     productFlavors {
         vanilla { // used for release and beta
-            buildConfigField "boolean", "AZTEC_EDITOR_AVAILABLE", "false"
```
Comment (maxme) [inline WordPress/build.gradle]: Should we drop `AZTEC_EDITOR_AVAILABLE` (always true) and `AppPrefs.isAztecEditorAvailable()` (always true)?

Comment (0nko) [inline WordPress/build.gradle]: I was going to do that in `develop` after the release. I just wanted to make it work for now. But I can do it all here if you prefer.

Comment (maxme) [inline WordPress/build.gradle]: That's fine, I was just wondering why we kept these.

**Comment by maxme** [comment]
- [ ] Aztec is the default editor, it should be opt-in.
- [ ] Uploaded media seem broken.
- [ ] No placeholder for title and content.

screenshot-2017-07-17_14 47 39 96

**Comment by 0nko** [comment]
The text hint bug still hasn't been fixed because there is a hidden character being inserted, which is currently being used by the parser and it's, therefore, not easy to remove. That's why it hasn't been fixed yet.

**Comment by maxme** [comment]
> The text hint bug still hasn't been fixed because there is a hidden character being inserted, which is currently being used by the parser and it's, therefore, not easy to remove. That's why it hasn't been fixed yet.

Ok, let me know when the PR is ready, it's the only thing remaining before 7.8 release.

**Comment by 0nko** [comment]
Image-loading is fixed and everything's ready.

**Comment by mzorz** [comment]
I'm testing media upload here

**Comment by maxme** [comment]
> Image-loading is fixed and everything's ready.

Image loading looks good.

Visual editor is the default, but there is no prompt to use the beta, is this the intended behavior?

**Comment by mzorz** [comment]
> Image loading looks good.ok confirmed image loading works again 👍

**Comment by 0nko** [comment]
The promo dialog should now behave as originally intended:

> Aztec enters open beta WPiOS 7.9 (July 17th release) / WPAndroid 7.7 (July 3rd release) hit the app stores. Users will see a pop-up alerting them to the new beta editor being available. - @astralbodies 

That is, after any user installs this version (be it a new installation or an update) the dialog is showed once. We might want to reverse this commit in `develop`.

**Comment by maxme** [comment]
I'm not sure to understand that last commit 583ead4, it's working fine without it (`EditorPromoRequired` defaults to `true`) and we should be super careful with the next release (might be a hotfix, 7.9 or develop) to avoid showing that dialog again.

**Comment by 0nko** [comment]
It was added so that users, who already saw the previous dialog (that means `EditorPromoRequired` would be `false` for them) will see this one, as well.

> ...we should be super careful with the next release (might be a hotfix, 7.9 or develop) to avoid showing that dialog again

That's why I mentioned that when merging back to `develop`, this commit should be reversed.

**Comment by maxme** [comment]
> It was added so that users, who already saw the previous dialog (that means EditorPromoRequired would be false for them) will see this one, as well.

But no one has [this](https://github.com/wordpress-mobile/WordPress-Android/blob/3fb70b4b73cc92f56cf4369ba0817d9aefe581de/WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java#L468) preference key set to true, excepted a few developers (and we don't care about them ;) ).

**Comment by 0nko** [comment]
Oh, I get it now.. It's a different preference key 🤦‍♂️ . I got it mixed up by the `setVisualEditorPromoRequired(false)` method call that used to be invoked for every new installation. I thought we just started reusing it and needed to reset it back to `true`. So you're right, it's redundant.

**Comment by maxme** [comment]
:shipit:

**Pull request merged**